aten/src/ATen/native/TensorShape.cpp

```diff
@@ -201,7 +204,8 @@ Tensor & _cat_out_cpu(Tensor& result, TensorList tensors, int64_t dim) {
 }
 
 Tensor _cat_cpu(TensorList tensors, int64_t dim) {
-  Tensor result = at::empty({0}, tensors[0].options());
+  ScalarType high_type = result_type(tensors);
+  Tensor result = at::empty({0}, tensors[0].options()).toType(high_type);
```

Comment:
nit: I think it's usually preferable to convert the `options` to the correct dtype, instead of the empty tensor.